src/platform_impl/redox/mod.rs

```diff
@@ -0,0 +1,1336 @@
+        Self {
+            user_events_sender: self.user_events_sender.clone(),
+            wake_socket: self.wake_socket.clone(),
+        }
+    }
+}
+
+impl<T> Unpin for EventLoopProxy<T> {}
+
+pub struct EventLoopWindowTarget<T: 'static> {
+    user_events_sender: mpsc::Sender<T>,
+    user_events_receiver: mpsc::Receiver<T>,
+    creates: Mutex<VecDeque<Arc<RedoxSocket>>>,
+    redraws: Arc<Mutex<VecDeque<WindowId>>>,
+    destroys: Arc<Mutex<VecDeque<WindowId>>>,
+    event_socket: Arc<RedoxSocket>,
+    wake_socket: Arc<TimeSocket>,
+}
+
+impl<T: 'static> EventLoopWindowTarget<T> {
+    pub fn primary_monitor(&self) -> Option<MonitorHandle> {
+        Some(MonitorHandle)
+    }
+
+    pub fn available_monitors(&self) -> VecDeque<MonitorHandle> {
+        let mut v = VecDeque::with_capacity(1);
+        v.push_back(MonitorHandle);
+        v
+    }
+
+    pub fn raw_display_handle(&self) -> RawDisplayHandle {
+        RawDisplayHandle::Orbital(OrbitalDisplayHandle::empty())
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct WindowId {
+    fd: u64,
+}
+
+impl WindowId {
+    pub const fn dummy() -> Self {
+        WindowId {
+            fd: u64::max_value(),
+        }
+    }
+}
+
+impl From<WindowId> for u64 {
+    fn from(id: WindowId) -> Self {
+        id.fd
+    }
+}
+
+impl From<u64> for WindowId {
+    fn from(fd: u64) -> Self {
+        Self { fd }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct DeviceId;
+
+impl DeviceId {
+    pub const fn dummy() -> Self {
+        DeviceId
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct PlatformSpecificWindowBuilderAttributes;
+
+struct WindowProperties<'a> {
+    flags: &'a str,
+    x: i32,
+    y: i32,
+    w: u32,
+    h: u32,
+    title: &'a str,
+}
+
+impl<'a> WindowProperties<'a> {
+    fn new(path: &'a str) -> Self {
+        // orbital:flags/x/y/w/h/t
+        let mut parts = path.splitn(6, '/');
+        let flags = parts.next().unwrap_or("");
+        let x = parts
+            .next()
+            .map_or(0, |part| part.parse::<i32>().unwrap_or(0));
+        let y = parts
+            .next()
+            .map_or(0, |part| part.parse::<i32>().unwrap_or(0));
+        let w = parts
+            .next()
+            .map_or(0, |part| part.parse::<u32>().unwrap_or(0));
+        let h = parts
+            .next()
+            .map_or(0, |part| part.parse::<u32>().unwrap_or(0));
+        let title = parts.next().unwrap_or("");
+        Self {
+            flags,
+            x,
+            y,
+            w,
+            h,
+            title,
+        }
+    }
+}
+
+impl<'a> fmt::Display for WindowProperties<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "orbital:{}/{}/{}/{}/{}/{}",
+            self.flags, self.x, self.y, self.w, self.h, self.title
+        )
+    }
+}
+
+pub struct Window {
```

Comment:
Could you split it into the separate file.

Reply:
Done